project/chapters/templates/chapters/chapter.html

```diff
@@ -8,4 +9,14 @@
 <h1>{{ page.title }}</h1>
 
 {{ page.introduction | richtext }}
-{% endblock content %}
\ No newline at end of file
+<form action="." method="post">
+    {% csrf_token %}
+    {% if member == False %}
+        <input id="membership_action" type="hidden" name="membership_action" value="join">
+        <input type="submit" class="btn btn-primary"value="{% translate 'Join group' %}">
+    {% else %}
+        <input id="membership_action" type="hidden" name="membership_action" value="leave">
+        <input type="submit" class="btn btn-secondary" value="{% translate 'Leave group' %}">
+    {% endif %}
+</form>
```

Comment:
To avoid confusion, anonymous users shouldn't see this form
```suggestion
{% if user.is_authenticated %}
<form action="." method="post">
    {% csrf_token %}
    {% if member == False %}
        <input id="membership_action" type="hidden" name="membership_action" value="join">
        <input type="submit" class="btn btn-primary"value="{% translate 'Join group' %}">
    {% else %}
        <input id="membership_action" type="hidden" name="membership_action" value="leave">
        <input type="submit" class="btn btn-secondary" value="{% translate 'Leave group' %}">
    {% endif %}
</form>
{% endif %}
```